### DISCUSSION TITLE
Trying to apply TorchMetrics on SpeakerID

### DISCUSSION THREAD

**Discussion by EddyGiusepe** [speechbrain/speechbrain#1675] [Q&A]
Hello everyone!
Can someone help me clarify the following:
Currently, I followed the tutorial and trained the [SpeakerIDfromScratch](https://colab.research.google.com/drive/1UwisnAjr8nQF3UnrkIJ4abBMAWzVwBMh?usp=sharing) task. 
I'm making some predictions with the trained model, plotting the LOSS curves, etc... But I'm also trying to calculate the [AUROC](https://torchmetrics.readthedocs.io/en/stable/classification/auroc.html) metric using Pytorch. I'm following an example of how to implement that, but when applying it to my task I'm receiving a Tensor with a value of zero. I did not expect that value, I expected a non-zero value.

![image](https://user-images.githubusercontent.com/69597971/182763028-ab880ca3-36b7-4d1b-ba66-42cf6c90da60.png)

Can someone help me on this?, please.

Or maybe I should ask is it possible to calculate this for this task?

I investigated, but I didn't find metrics for this Speaker Identification task

I'm following this [example](https://torchmetrics.readthedocs.io/en/stable/classification/auroc.html) (for the Multiclass case):
```
from torchmetrics.functional import auroc
preds = torch.tensor([[0.90, 0.05, 0.05],
                      [0.05, 0.90, 0.05],
                      [0.05, 0.05, 0.90],
                      [0.85, 0.05, 0.10],
                      [0.10, 0.10, 0.80]])
target = torch.tensor([0, 1, 1, 2, 2])
auroc(preds, target, num_classes=3)
```
 
Thank you very much!

**Comment by Yu-Miri** [comment]
Did you succeed in changing the metrics?
Trained speakers should predict speakers properly, and untrained speakers should predict unknown.
So if you are a trained speaker through AUROC, you are 'positive', and if you are an untrained speaker, I want to train something called 'negative', but I wonder if you succeeded.
